### DISCUSSION TITLE
Beginner How to

### DISCUSSION THREAD

**Discussion by hds1** [dirkwhoffmann/Moira#19] [Q&A]
I compiled the testRunner and i'am lost already.

I can't figure out how to use Moira with a custom BIOS Rom (Kiss-68030 in this case, SBC68030 Board).
Is there a bit more documentattion or a pointer to it of how to begin an emulator session with Moira ?

**Comment by dirkwhoffmann** [comment]
What do you want to achieve? Moira is just the CPU emulator. You will have to emulate the rest of the Board by yourself and exchange data with Moira by implementing the `read` and `write` functions. Also, Moira does not support the 68030 yet. Only models up to the 68020 can be emulated.

**Comment by hds1** [comment]
Am Sonntag, dem 20.11.2022 um 22:08 -0800 schrieb Dirk Hoffmann:
> 
> What do you want to achieve? Moira is just the CPU emulator.
Precisely. If you want to emulate a simple SBC 68030 board, say testing
your BIOS, you want an single-stepping-tracing CPU Emulator.
READ, WRITE functions for the Board i.e. I/O are quite simple.
> You will have to emulate the rest of the Board by yourself and exchange
> data with Moira by implementing the `read` and `write` functions. Also,
> Moira does not support the 68030 yet. Only models up to the 68020 can be
> emulated.
Ok.

**Comment by dirkwhoffmann** [comment]
I guess it's no longer relevant, because you need 68030 emulation, but I've just added [some more documentation](https://dirkwhoffmann.github.io/Moira/docs/Tutorials/UsingMoiraInYourOwnApp.html) about how to use Moira in another application.

**Comment by hds1** [comment]
I wouldnt say it is not relevant. Moira is a new program.

It would be nice to attach your own machine (how simple it might be, like in Musashi) to the CPU emulator.
In that respect Moira would grow to a more universal machine emulator and not only an Atari/Amiga emulator.

But i cannot oversee how much work it would be to seperate CPU Emulation from the Hardware Description part.






Am Dienstag, dem 22.11.2022 um 08:01 -0800 schrieb Dirk Hoffmann:

**Comment by dirkwhoffmann** [comment]
Musashi and Moira are pretty much exchangeable. Only the API names differ.

**Comment by hds1** [comment]
Great to hear.
But Moira has the possibility of a nicer Interface for debugging purposes (tracing, single stepping) via vAmiga ?





Am Dienstag, dem 22.11.2022 um 08:27 -0800 schrieb Dirk Hoffmann:

**Reply by dirkwhoffmann** [reply]
Single stepping, breakpoints etc. are all [Moira features](https://dirkwhoffmann.github.io/Moira/docs/HowTo/HowToUseTheDebugger.html). vAmiga only adds a nice GUI on top.

**Comment by hds1** [comment]
Even better. What i have in mind is basic BIOS development for (simple, medium complex) SBC with 68030.

So lets the docs rolling ....


Am Dienstag, dem 22.11.2022 um 08:34 -0800 schrieb Dirk Hoffmann:
